### PR TITLE
[IMP] event: redesign badges with multiple formats

### PR DESCRIPTION
**Pull request by Odoonan**
Before, event badges were an a4 page divided in 4
quarters, meant to be foldable into a single badge.
It contained folding instructions, event instructions
and was white mostly everywhere.

We want to introduce some customization to allow
different colors between ticket types and a background
image.

Therefore, we introduce 3 different formats to print badges
(also sent to attendees through event communication)
- a4 foldable (similar to previous one but redesigned)
- a6
- a4 4 per page

The templates are intensively redesigned to a new style.
New templates are introduced to match new formats.
Example tickets are still managed, printing an appropriate
example ticket according to the event format.
Printing attendees from a list view with different events
will use a6 instead of 4 per page format.
Printing several event example tickets will still use each
event format appropriately

Models:

- event.event receives a selection field badge_format
and an image field badge_image. This will be printed
on the background of badge cards.

- event.event.ticket receives a color char field. It
can be picked from the event form view with a color picker,
and will be the backgroud of a band on the bottom of badge cards.

Renaming:

Scss file and templates are renamed from "_foldable_badge"
to "_badge", as now badges are not always foldable. Namespaces
of selectors follow the same logic. We still have specific
styling for the foldable version so there are still selectors
using a 'foldable' name.

Task-3473293

### CODE REVIEW

File: addons/event/models/event_event.py [hunk 1]
```diff
@@ -224,10 +224,19 @@ def _lang_get(self):
     lang = fields.Selection(_lang_get, string='Language',
         help="All the communication emails sent to attendees will be translated in this language.")
     # ticket reports
+    badge_format = fields.Selection(
+        string='Badge Dimension',
+        selection=[
+            ('A4_foldable', 'A4 foldable'),
```
Comment (tde-banana-odoo) [inline addons/event/models/event_event.py:230]: Seems it is 'A4_half_fold', according to "The Internet'

File: addons/event/models/event_event.py [hunk 2]
```diff
@@ -224,10 +224,19 @@ def _lang_get(self):
...
+            ('A6', 'A6'),
+            ('four_per_sheet', '4 per sheet')
```
Comment (tde-banana-odoo) [inline addons/event/models/event_event.py:232]: Seems it is 'A4_french_fold', according to "The Internet'

File: addons/event/models/event_event.py [hunk 3]
```diff
@@ -224,10 +224,19 @@ def _lang_get(self):
...
+        ], default='A6')
+    badge_image = fields.Image('Badge Background', max_width=1024, max_height=1024)
     ticket_instructions = fields.Html('Ticket Instructions', translate=True,
         compute='_compute_ticket_instructions', store=True, readonly=False,
         help="This information will be printed on your tickets.")
 
+
```
Comment (tde-banana-odoo) [inline addons/event/models/event_event.py:239]:
```suggestion
```

File: addons/event/models/event_ticket.py [hunk 4]
```diff
@@ -80,6 +80,8 @@ def default_get(self, fields):
     seats_used = fields.Integer(string='Used Seats', compute='_compute_seats', store=False)
     is_sold_out = fields.Boolean(
         'Sold Out', compute='_compute_is_sold_out', help='Whether seats are not available for this ticket.')
+    # reports
+    color = fields.Char('Color', default="#875A7B")
```
Comment (tde-banana-odoo) [inline addons/event/models/event_ticket.py:84]: Do we need a custom color, in addition to report colors and company email colors :D ?

File: addons/event/report/event_event_templates.xml [hunk 5]
```diff
@@ -1,108 +1,135 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
-<!-- EVENT FOLDABLE BADGE -->
+<!-- EVENT BADGE CARD -->
+
+<template id="event_report_template_badge_card">
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:6]: As it is a tool template, would move at bottom of the file maybe, to keep the main ones on top ?

**Review by tde-banana-odoo — COMMENTED** [review]
Just saw some light :D

**Review by Odoonan — COMMENTED** [review]
@tde-banana-odoo Pong! Answered to your review :) I also renamed - maybe while you were reviewing - a few records from _foldable_badge to _badge. It will probably require a small upgrade script, but file exists already so while we are here...

Cheers!

File: addons/event/models/event_event.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/event/models/event_event.py:232]: Relooking at various website, seems the *french fold* is 4 / A4 (fold in 4 a bit like we do generally, not like a promotional folder or stuff like that) and *half fold* is 2 / A4 .

Note that we could rename *french fold* to *french fold :wine_glass: * .

File: addons/event/report/event_event_templates.xml [hunk 6]
```diff
@@ -218,6 +219,32 @@
...
+                <h4 t-if="attendee and attendee.company_id" class="o_event_badge_font_faded" t-field="attendee.company_id.name"/>
+                <h4 t-elif="not attendee"><span class="o_event_badge_font_faded">My Placeholder Company</span></h4>
+            </div>
+            <div class="position-absolute bottom-0 w-100 text-center">
+                <img t-if="event.organizer_id.image_256" class="o_event_badge_logo text-center" t-att-src="image_data_uri(event.organizer_id.image_256)"/>
+                <span t-if="event.badge_format != 'A4_french_fold'" class="o_event_badge_barcode_container"></span>
+                <t t-set="first_ticket" t-value="event.event_ticket_ids[0] if event.event_ticket_ids else None"/>
+                <t t-set="ticket" t-value="attendee.event_ticket_id if attendee else first_ticket if first_ticket else None"/>
+                <div class="text-center mt-2 w-100" t-attf-style="background-color: {{ticket.color if ticket and ticket.color else '#875A7B'}};">
+                    <div t-if="ticket" class="p-3 fs-3" t-out="ticket.name"/>
```
Comment (std-odoo) [inline addons/event/report/event_event_templates.xml:241]: shouldn't the `t-if="ticket"` be on the parent div ?

(when there's no ticket, badge on the right the div is not visible anyway)

image

**Review by std-odoo — COMMENTED** [review]
@Odoonan Hi :) 

I have a few comments / suggestions

Cheers

File: addons/event/report/event_event_templates.xml [hunk 7]
```diff
@@ -218,6 +219,32 @@
     </t>
 </template>
 
+<!-- EVENT BADGE CARD (tool template used in all badge templates)-->
+
+<template id="event_report_template_badge_card">
+    <t t-set="badge_image_url" t-value="'data:image/png;base64,%s' % event.badge_image.decode('utf-8') if event.badge_image else ''"/>
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/event/report/event_event_templates.xml:225]: Actually when looking at the code of image data uri, it does pretty much the same :p I think I found them as is in the reporting tools templates at the time. Changed :)

File: addons/event/models/event_event.py [hunk 8]
```diff
@@ -224,6 +224,14 @@ def _lang_get(self):
     lang = fields.Selection(_lang_get, string='Language',
         help="All the communication emails sent to attendees will be translated in this language.")
     # ticket reports
+    badge_format = fields.Selection(
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/event/models/event_event.py:227]: We could, is the default not sufficient in terms of migration? :)

**Comment by Odoonan** [comment]
Oy @std-odoo , thanks for the review, I've applied the changes. Still have to make badge format required -> I am not sure: is default value not enough in terms of migration?

Cheers!

File: addons/event/models/event_event.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/event/models/event_event.py:227]: Alright, done :)

File: addons/event/models/event_ticket.py (hunk 4 above)
Comment (KangOl) [inline addons/event/models/event_ticket.py:84]: Real bikeshedding: Use color of the community version
```suggestion
    color = fields.Char('Color', default="#71639E")
```

Comment (KangOl) [reply addons/event/models/event_ticket.py]: That was partly a joke, but then you could change the default value in the `event_enterprise` module to match the enterprise theme.

Comment (Odoonan) [reply addons/event/models/event_ticket.py]: I get it :facepalm: :zzz: , I would rather go with previous color then, as we used a hardcoded background image with that color - not in community version theme - until last stable version. That way we keep something close to what we had. Also, it avoids creating a file just for setting that default color :p

**Pull request closed without merging**